api/v1alpha1/jobset_webhook.go

```diff
@@ -96,8 +96,8 @@ func (js *JobSet) ValidateUpdate(old runtime.Object) error {
 		}
 	}
 
-	if !apiequality.Semantic.DeepEqual(oldSpec, *mungedSpec) {
-		specDiff := cmp.Diff(oldSpec, *mungedSpec)
+	if !apiequality.Semantic.DeepEqual(oldSpec.ReplicatedJobs, mungedSpec.ReplicatedJobs) {
+		specDiff := cmp.Diff(oldSpec.ReplicatedJobs, mungedSpec.ReplicatedJobs)
 		return field.Forbidden(field.NewPath("spec"), fmt.Sprintf("jobset updates may not change fields other than %s\n%v", strings.Join(updatableFields, ","), specDiff))
 	}
 	return nil
```

Comment:
```suggestion
  allErrs = append(allErrs, apivalidation.ValidateImmutableField(mungedSpec.ReplicatedJobs, oldSpec.ReplicatedJobs, specPath.Child("replicatedJobs"))...)
  allErrs = append(allErrs, apivalidation.ValidateImmutableField(mungedSpec.SuccessPolicy, oldSpec.SuccessPolicy, specPath.Child("successPolicy"))...)
  allErrs = append(allErrs, apivalidation.ValidateImmutableField(mungedSpec.FailurePolicy, oldSpec.FailurePolicy, specPath.Child("failurePolicy"))...)
return allErrs   
```

Reply:
and import `apivalidation "k8s.io/apimachinery/pkg/api/validation"`

Reply:
they are immutable. See on jobset_types.go line 53 and 60

Reply:
I can add some tests to demonstrate that....

Reply:
Conclusion:
After some tests, FailurePolicy is immutable only if it was set before. Should I do something else on this side?

Reply:
ah, right, I updated the suggestion, lets still use `apivalidation.ValidateImmutableField` to check the replicatedJobs field and have a comment indicating that success and failure policy are made immutable via CEL

Reply:
done